src/Autofac.Extras.MvvmCross/AutofacMvxIocProvider.cs

```diff
@@ -387,7 +403,8 @@ public virtual void RegisterSingleton(Type tInterface, object theObject)
 
             // You can't inject properties on a pre-constructed instance.
             cb.RegisterInstance(theObject).As(tInterface).AsSelf().SingleInstance();
-            cb.Update(this.Container);
+
+            cb.Update(Container);
```

Comment:
The reason is that you register instances in MvvmCross using "Mvx.IocRegisterSingleton".
So you never really see the underlying DI frameworks internals which is quite good I think